src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -639,13 +639,17 @@ public async Task WaitAsyncForSelfTerminatingChild()
         [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
         public async Task WaitAsyncForProcess()
         {
-            Process p = CreateSleepProcess(WaitInMS);
+            Process p = CreateProcessPortable(RemotelyInvokable.SleepWithWrite, "1000");
+            p.StartInfo.RedirectStandardOutput = true;
```

Comment:
I am not able to figure out what this is trying to fix and why it works. Could you please elaborate?

I believe that this fix is not quite right. If the stdout buffer is small or non-existing, the test will hang. The process will be stuck on writing to stdout. This write will never end because of `ReadToEnd` that reads from other side of the pipe is done only after the process exits.

Reply:
As written, WaitAsyncForProcess takes at least 30 seconds to complete. That is because it waits on the first to finish of: a process that sleeps for 30 seconds, and a 60 second delay. The other tests are mostly finished by the time it starts, so it causes the test run to wait that long. (Although, having mitigated that, there is a subsequent 5 minute delay as mentioned above)

If this delay was intentional I guess it was to try to avoid the process exiting before waiting on it to exit. So, I was hoping to find a way to block the child process from exiting until `WaitForExitAsync()` is called.

Reply:
Can we make the test outer loop instead?

Reply:
Perhaps I can use an event such as in WaitForSignal().

Reply:
Sure, I'll do that for now.

Reply:
Done, and opened issue.